airbyte-integrations/connectors/source-file/source_file/spec.json

```diff
@@ -6,22 +6,31 @@
     "title": "File Source Spec",
     "type": "object",
     "additionalProperties": false,
-    "required": ["dataset_name", "format", "url", "provider"],
+    "required": ["archive_format", "file_format", "url", "provider"],
     "properties": {
-      "dataset_name": {
+      "archive_format": {
         "type": "string",
-        "description": "Name of the final table where to replicate this file (should include only letters, numbers dash and underscores)"
+        "enum": ["Not Archived", "tar.gz"],
+        "default": "Not Archived",
+        "description": "File Format of the archive file"
       },
-      "format": {
+      "file_format": {
         "type": "string",
-        "enum": ["csv", "json", "jsonl", "excel", "feather", "parquet"],
+        "description": "Valid JSON string describing the File Format of (extracted) file(s). Allowed formats: [\"csv\", \"json\", \"jsonl\", \"excel\", \"feather\", \"parquet\"] (Warning: some format may be experimental, please refer to docs).",
         "default": "csv",
-        "description": "File Format of the file to be replicated (Warning: some format may be experimental, please refer to docs)."
+        "examples": ["csv", "json", "{\"file1.xyz\": \"csv\", \"file2.xyz\": \"json\", \"file3.xyz\": \"jsonl\"}"]
+      },
+      "file_rename": {
+        "type": "string",
+        "description": "Valid JSON/string describing how to rename downloaded file(s). This string determines the name of the destination table for each file",
+        "default": "default_tablename",
+        "examples": ["single_file_table_name", "{\"file1.xyz\": \"tablename1\", \"file2.xyz\": \"tablename2\"}"]
       },
       "reader_options": {
         "type": "string",
+        "default": "{}",
         "description": "This should be a valid JSON string used by each reader/parser to provide additional options and tune its behavior",
-        "examples": ["{}", "{'sep': ' '}"]
+        "examples": ["{\"all\": {\"sep\": \" \"}}", "{\"file1.xyz\": {\"sep\": \" \"}, \"file2.xyz\": {\"sep\": \"|\"}}"]
       },
```

Comment:
I think that tweaking the specs as you did leads to a less strict validation of parameters that can lead to more error for users.
What do you think of changing the spec structure to have a more user friendly interface such as:
```json
{
    "documentationUrl": "https://docs.airbyte.io/integrations/sources/file",
    "connectionSpecification": {
        "$schema": "http://json-schema.org/draft-07/schema#",
        "title": "File Source Spec",
        "type": "object",
        "additionalProperties": false,
        "required": ["url", "provider", "is_archive", "files"],
        "properties": {
            "url": {
                "type": "string",
                "description": "URL path to access the file to be replicated"
            },
            "is_archive": {
              "type": "boolean",
              "description": "Wether the source file is a tar archive"
            },
            "provider": {},
            "files": {
                "type": "array",
                "description": "List of files to download, set multiple files if the source is an archive",
                "items": {
                    "type": "object",
                    "additionalProperties": false,
                    "required": [
                        "format",
                        "dataset_name"
                    ],
                    "properties": {
                        "format": {
                            "type": "string",
                            "enum": ["csv", "json", "jsonl", "excel", "feather", "parquet"
                            ],
                            "default": "csv",
                            "description": "File Format of the file to be replicated (Warning: some format may be experimental, please refer to docs)."
                        },
                        "dataset_name": {
                            "type": "string",
                            "description": "Name of the final table where to replicate this file (should include only letters, numbers dash and underscores)"
                        },
                        "reader_options": {
                            "type": "string",
                            "description": "This should be a valid JSON string used by each reader/parser to provide additional options and tune its behavior",
                            "examples": ["{}", "{'sep': ' '}"]
                        }
                    }
                }
            }
        }
    }
}
```